Mathlib/LinearAlgebra/Matrix/DotProduct.lean

```diff
@@ -74,30 +72,79 @@ end Semiring
   (Finset.sum_eq_zero_iff_of_nonneg fun i _ => mul_self_nonneg (v i)).trans <| by
     simp [Function.funext_iff]
 #align matrix.dot_product_self_eq_zero Matrix.dotProduct_self_eq_zero
 
+section StarOrderedRing
+
+variable [PartialOrder R] [NonUnitalRing R] [StarOrderedRing R] [NoZeroDivisors R]
+
 /-- Note that this applies to `ℂ` via `Complex.strictOrderedCommRing`. -/
 @[simp]
-theorem dotProduct_star_self_eq_zero [PartialOrder R] [NonUnitalRing R] [StarOrderedRing R]
-    [NoZeroDivisors R] {v : n → R} : dotProduct (star v) v = 0 ↔ v = 0 :=
+theorem dotProduct_star_self_eq_zero {v : n → R} : dotProduct (star v) v = 0 ↔ v = 0 :=
   (Finset.sum_eq_zero_iff_of_nonneg fun i _ => (@star_mul_self_nonneg _ _ _ _ (v i) : _)).trans <|
     by simp [Function.funext_iff, mul_eq_zero]
 #align matrix.dot_product_star_self_eq_zero Matrix.dotProduct_star_self_eq_zero
 
 /-- Note that this applies to `ℂ` via `Complex.strictOrderedCommRing`. -/
 @[simp]
-theorem dotProduct_self_star_eq_zero [PartialOrder R] [NonUnitalRing R] [StarOrderedRing R]
-    [NoZeroDivisors R] {v : n → R} : dotProduct v (star v) = 0 ↔ v = 0 :=
+theorem dotProduct_self_star_eq_zero {v : n → R} : dotProduct v (star v) = 0 ↔ v = 0 :=
   (Finset.sum_eq_zero_iff_of_nonneg fun i _ => (@star_mul_self_nonneg' _ _ _ _ (v i) : _)).trans <|
     by simp [Function.funext_iff, mul_eq_zero]
 #align matrix.dot_product_self_star_eq_zero Matrix.dotProduct_self_star_eq_zero
 
+@[simp]
+lemma conjTranspose_mul_self_eq_zero_iff {A : Matrix m n R} : (Aᴴ * A) = 0 ↔ A = 0 :=
+  ⟨fun h => Matrix.ext fun i j =>
+    (congr_fun <| dotProduct_star_self_eq_zero.1 <| Matrix.ext_iff.2 h j j) i,
+  fun h => h ▸ Matrix.mul_zero _⟩
+
+@[simp]
+lemma self_mul_conjTranspose_eq_zero_iff {A : Matrix m n R} : (A * Aᴴ) = 0 ↔ A = 0 :=
+  ⟨fun h => Matrix.ext fun i j =>
+    (congr_fun <| dotProduct_self_star_eq_zero.1 <| Matrix.ext_iff.2 h i i) j,
+  fun h => h ▸ Matrix.zero_mul _⟩
+
+lemma conjTranspose_mul_self_mul_eq_zero_iff  (A : Matrix m n R) (B : Matrix n p R) :
+    (Aᴴ * A) * B = 0 ↔ A * B = 0 := by
+  refine' ⟨ fun h => _, fun h => by simp only [Matrix.mul_assoc, h, Matrix.mul_zero] ⟩
+  apply_fun (fun x => Bᴴ * x) at h
+  rw [Matrix.mul_zero, Matrix.mul_assoc, ← Matrix.mul_assoc, ← conjTranspose_mul] at h
+  exact conjTranspose_mul_self_eq_zero_iff.1 h
+
+lemma self_mul_conjTranspose_mul_eq_zero_iff (A : Matrix m n R) (B : Matrix m p R) :
+    (A * Aᴴ) * B = 0 ↔ Aᴴ * B = 0 := by
+  simpa only [conjTranspose_conjTranspose] using conjTranspose_mul_self_mul_eq_zero_iff  Aᴴ _
+
+lemma conjTranspose_mul_self_mulVec_eq_zero_iff (A : Matrix m n R) (v : n → R) :
+    (Aᴴ * A).mulVec v = 0 ↔ A.mulVec v = 0 := by
+  rw [← mulVec_mulVec]
+  refine ⟨ fun h => ?_ , fun h => by simp only [h, mulVec_zero] ⟩
+  apply_fun (fun x => dotProduct (star v) x) at h
+  rwa [dotProduct_mulVec, dotProduct_zero, vecMul_conjTranspose, star_star,
+    dotProduct_star_self_eq_zero] at h
```

Comment:
I think #6614 should make it possible to solve this with a single `simpa` line, by appealing to `self_mul_conjTranspose_mul_eq_zero_iff` with `B = col v`.